pkg/issuer/acme/dns/route53/route53.go

```diff
@@ -83,17 +85,27 @@ func NewDNSProvider() (*DNSProvider, error) {
 // NewDNSProviderAccessKey returns a DNSProvider instance configured for the AWS
 // Route 53 service using static credentials from its parameters
 func NewDNSProviderAccessKey(accessKeyID, secretAccessKey, hostedZoneID, region string) (*DNSProvider, error) {
-
-	creds := credentials.NewStaticCredentials(accessKeyID, secretAccessKey, "")
-
 	r := customRetryer{}
 	r.NumMaxRetries = maxRetries
 
-	config := request.WithRetryer(aws.NewConfig(), r).WithCredentials(creds)
+	config := request.WithRetryer(aws.NewConfig(), r)
 
 	if region != "" {
 		config.WithRegion(region)
 	}
+
+	creds := credentials.NewChainCredentials([]credentials.Provider{
+		&credentials.EnvProvider{},
+		&ec2rolecreds.EC2RoleProvider{
+			Client: ec2metadata.New(session.New(config)),
+		},
+		&credentials.StaticProvider{Value: credentials.Value{
+			AccessKeyID:     accessKeyID,
+			SecretAccessKey: secretAccessKey,
+		}},
```

Comment:
From looking at the source for the ChainCredentials provider, it seems that credentials are tried in order from start to finish. If a user is running their cluster in AWS, would this not cause the manually specified credentials to be ignored as the metadata service *is* available (despite the node not actually having sufficient IAM privileges to perform the later operations)?

Reply:
Ah, good point I've got the ordering mixed up here. Will fix.